package ch05.ex03.case02;

import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.IOException;

public class FileOutputStreamTest02 {
	public static void main(String[] args) throws FileNotFoundException {
		// byte[] 배열에 A-Z까지(26개) 넣고 그 배열을 한꺼번에 파일에 쓰기
		FileOutputStream fos = new FileOutputStream("output.txt");		
		
		try (fos) { 
			fos.write(65); // A
			fos.write(66); // B
			fos.write(67); // C
		} catch (FileNotFoundException e) {
			e.printStackTrace();
		} catch (IOException e) { // fos.close() 리소스 해제
			e.printStackTrace();
		}
		
		System.out.println("파일 생성과 출력이 완료되었습니다.");
	}
}

// try-with-resources
// close() 메소드 자동 실행. 그래서 IOException 처리를 finally로 번거롭게 안 해줘도 됨